infra/charts/feast-feature-server/templates/deployment.yaml

```diff
@@ -33,19 +33,46 @@ spec:
           env:
             - name: FEATURE_STORE_YAML_BASE64
               value: {{ .Values.feature_store_yaml_base64 }}
-          command: ["feast", "serve", "-h", "0.0.0.0"]
+          command:
+            {{- if eq .Values.feast_mode "offline" }}
+            - "feast"
+            - "serve_offline"
+            - "-h"
+            - "0.0.0.0"
+            {{- else if eq .Values.feast_mode "ui" }}
+            - "feast"
+            - "ui"
+            - "-h"
+            - "0.0.0.0"
+            {{- else if eq .Values.feast_mode "registry" }}
+            - "feast"
+            - "serve_registry"
+            {{- else }}
+            - "feast"
+            - "serve"
+            - "-h"
+            - "0.0.0.0"
+            {{- end }}
           ports:
-            - name: http
+            - name: {{ .Values.feast_mode | default "online" }}
```

Comment:
you shouldn't have to set `default "online"` as you've already set it to default to "online" in `values.yaml`